### PR TITLE
Enable endpoint deletion

### LINKED ISSUE #252: Enable deletion of an endpoint
Currently users cannot delete an endpoint. I'd see this as a very negative aspect of the User Experience, ability to undo and do-over are helpful comforts when dealing with something new and I think the considerations that open up from what Bothan affords create a user environment where that comfort would be desired.

It'd be good to know if the lack of a delete function is feature or bug? Is there a design principle guiding why this isn't yet part of Bothan @Floppy & @pezholio?

### PR DESCRIPTION
**Pull request by quadrophobiac**
This PR fixes #252

Changes proposed in this pull request:

- changes to index page to conditionally display edit metadata and delete metrics hyperlinks
- DELETE route for metrics;
- addition of tests to confirm delete route works and admin features display only to authenticated

I really wanted to include features that mocked the user interaction with these changes as I saw the addition of a DELETE route as primarily a usability enhancement. However I don't it is possible to do so. The capybara-webkit steps which worked for dashboards here fail to return a page displaying the authenticated admin functions, whereas basic auth signin does (the inverse was true for the dashboard features in PR #270)

I'd like feedback if the test coverage included in this PR is sufficient for the changes made to the application

### CODE REVIEW

File: .byebug_history [hunk 1]
```diff
@@ -0,0 +1 @@
+@auth.provided? and @auth.basic? and @auth.credentials and @auth.credentials == [ENV['METRICS_API_USERNAME'], ENV['METRICS_API_PASSWORD']]
```
Comment (Floppy) [inline .byebug_history]: ignore this file

File: lib/bothan/metrics.rb [hunk 2]
```diff
@@ -114,6 +114,16 @@ def self.registered(app)
         get_metric_range(params)
       end
 
+      app.delete '/metrics/:metric' do
+        protected!
+        Metric.where(name: params[:metric]).each do |metric|
```
Comment (Floppy) [inline lib/bothan/metrics.rb]: Best way to do this is use https://apidock.com/rails/ActiveRecord/Relation/destroy_all, which will do it (a) in a single line and (b) do all the callbacks and associations.

**Comment by Floppy** [comment]
It might be something for another PR after this is merged, but it would be useful to delete particular data elements as well, not just an entire metric.

File: lib/bothan/metrics.rb (hunk 2 above)
Comment (quadrophobiac) [reply lib/bothan/metrics.rb]: cheers, thanks for the pointer on this I sensed that my way was a bit of a kludge

File: .byebug_history (hunk 1 above)
Comment (quadrophobiac) [reply .byebug_history]: it's already in the gitignore but it won't go away -https://github.com/theodi/bothan/blob/master/.gitignore#L14
 I must have committed it by accident at some point, hope the force delete purges it

**Pull request merged**